feat: update CI/CD documentation to use service accounts (#235)
CerebriumAI/documentation: cerebrium/deployments/ci-cd.mdx

---
title: "CI/CD Pipelines"
description: "Automate your Cerebrium deployments using GitHub Actions"
---

If you would like to automatically deploy a new version of your app to production/development when you push a branch or press a button you should configure a Continuous Integration or Continuous Deployment system (CI/CD).

This guide sets up a CI/CD system using GitHub Actions and Cerebriums' Service Accounts.

> We recommend maintaining **separate development and production apps** in separate projects in Cerebrium so that changes can be safely tested before going live.

### 1. Authenticating to Cerebrium

> CI/CD workflows use **machine credentials**, not personal user credentials. Cerebrium Service Accounts are designed for this. Create separate service accounts for your production and development projects.

Reviewer: Should be explicit about what the difference between machine and user credentials are here. Want to maintain as few concepts as possible so that users don't get confused. Use exactly the terminology we use in the platform.

Author: Reworded a bit

1. Go to the [Cerebrium Dashboard](https://dashboard.cerebrium.ai/) and open the **API Keys** page.
2. Click **Create Service Account**, name it `"GitHub Actions CI/CD"`, choose an expiry date, and click **Create**.

> 3. **Copy the token** generated for your service account.

Reviewer: ```suggestion
3. **Copy the token** generated for the new service account.
```

![Cerebrium API Keys dashboard](/images/api-key-sa.png)

> ### 2. Define secrets in your GitHub environment

Reviewer: ```suggestion
### 2. Define secrets in a GitHub environment
```

> Now, store this token in your GitHub repository for use in your GitHub Actions workflows.

Reviewer: ```suggestion
Store this token in a secret for use in GitHub Actions workflows.
```

> 1. Navigate to your **GitHub repository → Settings → Environments**
> 2. Create environments named `"dev"` and/or `"prod"` (depending on your needs).
> 3. Inside each environment, add a secret:
>    - **Name**: `CEREBRIUM_SERVICE_ACCOUNT_TOKEN`
>    - **Value**: the token you copied in the previous step.

Reviewer: ```suggestion
1. Navigate to the **GitHub repository → Settings → Environments**
2. Create environments named `"dev"` and/or `"prod"` (Adjust to specific needs)
3. Inside each environment, add a secret:
   - **Name**: `CEREBRIUM_SERVICE_ACCOUNT_TOKEN`
   - **Value**: the token copied in the previous step
```

> 🔒 Use _secrets_, not plain environment variables, to prevent leaking sensitive data in logs.

![GitHub](/images/githubActions.png)

### 3. GitHub Actions Workflow

GitHub Actions uses workflow files located in the `.github/workflows/` directory of your repository. You’ll need to create one to define your deployment process.

> 1. In the root of your GitHub repository, create the directory `.github/workflows` if it doesn’t already exist.
> 2. Inside that directory, create a new file named `cerebrium-deploy.yml`.
> 3. Paste the following YAML into the file:

Reviewer: ```suggestion
1. Create a `.github/workflows`  directory in the root of a new or existing GitHub repository
2. Within that directory, create a new file named `cerebrium-deploy.yml`
3. Paste the following YAML into the file:
```
